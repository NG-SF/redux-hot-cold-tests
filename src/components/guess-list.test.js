import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import {GuessList} from './guess-list';

describe('<GuessList />', () => {
  const guesses = [];
  beforeAll(() => {
    for (let i = 0; i < 10; i++) {
          guesses.push(`Guess ${i}`);
        }
    });
    it('Renders without crashing', () => {
        shallow(<GuessList guesses={guesses} />);
    });

    it('Should match its snapshot', () => {
    const tree = renderer.create(<GuessList guesses={guesses}/>).toJSON();
    expect(tree).toMatchSnapshot();
    });

    it('Renders a list of guesses', () => {
        const guesses = [3, 42, 77];
        const wrapper = shallow(<GuessList guesses={guesses} />);
        const items = wrapper.find('li');
        expect(items.length).toEqual(guesses.length);
        guesses.forEach((guess, index) => {
            expect(items.at(index).text()).toEqual(guess.toString());
        });
    });
});