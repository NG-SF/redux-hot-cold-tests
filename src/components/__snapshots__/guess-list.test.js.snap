// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GuessList /> Should match its snapshot 1`] = `
<ul
  className="guessBox clearfix"
  id="guessList"
>
  <li>
    Guess 0
  </li>
  <li>
    Guess 1
  </li>
  <li>
    Guess 2
  </li>
  <li>
    Guess 3
  </li>
  <li>
    Guess 4
  </li>
  <li>
    Guess 5
  </li>
  <li>
    Guess 6
  </li>
  <li>
    Guess 7
  </li>
  <li>
    Guess 8
  </li>
  <li>
    Guess 9
  </li>
</ul>
`;
